src/wasm/wat-parser-internal.h

```diff
@@ -372,6 +372,72 @@ std::optional<LexIntResult> integer(std::string_view in) {
   return {};
 }
 
+// idchar ::= '0' | ... | '9'
+//          | 'A' | ... | 'Z'
+//          | 'a' | ... | 'z'
+//          | '!' | '#' | '$' | '%' | '&' | ''' | '*' | '+'
+//          | '-' | '.' | '/' | ':' | '<' | '=' | '>' | '?'
+//          | '@' | '\' | '^' | '_' | '`' | '|' | '~'
+std::optional<LexResult> idchar(std::string_view in) {
+  LexCtx ctx(in);
+  if (ctx.empty()) {
+    return {};
+  }
+  char c = ctx.next()[0];
```

Comment:
Would it be better to create a method like `peek(int)`, which checks the string without consuming it?

Reply:
Good idea, I'll add that.